main.cpp

```diff
@@ -124,379 +124,157 @@ struct IntType;
-    {
-        
-    }
-    ~DoubleType()
-    {
-        delete value;
-    }
-
-    DoubleType& add( double operand )
-    {
-        if (value != nullptr)    
-            *value += operand;
-
-        return *this;
-    }
+    FloatType(float _value) : value(new float(_value)) { }
+    ~FloatType() { delete value; }
 
-    DoubleType& add(const FloatType& ft);
-    DoubleType& add(const DoubleType& ft);
-    DoubleType& add(const IntType& ft);
-
-    DoubleType& subtract( double operand )
-    {
-        if (value != nullptr)   
-            *value -= operand;
-
-        return *this;
-    }
-
-    DoubleType& subtract(const FloatType& ft);
-    DoubleType& subtract(const DoubleType& ft);
-    DoubleType& subtract(const IntType& ft);
-
-    DoubleType& multiply( double operand )
-    {
-        if (value != nullptr) 
-            *value *= operand;
-
-        return *this;
-    }
-
-    DoubleType& multiply(const FloatType& ft);
-    DoubleType& multiply(const DoubleType& ft);
-    DoubleType& multiply(const IntType& ft);
-    
-
-    DoubleType& divide( double operand )
-    {
-        if (operand == 0.0)
-            std::cout << "warning: floating point division by zero!" << std::endl;
-
-        if (value != nullptr)
-            *value /= operand;
-
-        return *this;
-    }
-
-    DoubleType& divide(const FloatType& ft);
-    DoubleType& divide(const DoubleType& ft);
-    DoubleType& divide(const IntType& ft);
-};
-
-struct IntType
-{
-    int* value = nullptr;
-
-    IntType(int _value) : value(new int(_value))
-    {
-    
-    }
-    ~IntType()
-    {
-        delete value;
-    }
-
-    IntType& add( int operand )
-    {
-        if (value != nullptr)    
-            *value += operand;
-
-        return *this;
-    }
-
-    IntType& add(const FloatType& ft);
-    IntType& add(const DoubleType& ft);
-    IntType& add(const IntType& ft);
-
-    IntType& subtract( int operand )
-    {
-        if (value != nullptr)   
-            *value -= operand;
-
-        return *this;
-    }
-
-    IntType& subtract(const FloatType& ft);
-    IntType& subtract(const DoubleType& ft);
-    IntType& subtract(const IntType& ft);
-
-    IntType& multiply( int operand )
-    {
-        if (value != nullptr) 
-            *value *= operand;
-
-        return *this;
-    }
+    FloatType& add( float operand );
+    FloatType& subtract( float operand );
+    FloatType& multiply( float operand );
+    FloatType& divide( float operand );
 
-    IntType& multiply(const FloatType& ft);
-    IntType& multiply(const DoubleType& ft);
-    IntType& multiply(const IntType& ft);
-    
-    IntType& divide( int operand )
-    {
-        if (operand != 0)
-        {
-            *value /= operand;    
-        }
-        else
-        {
-            std::cout << "error: integer division by zero is an error and will crash the program!" << std::endl ;
-        }
-        return *this;
-    }
+    operator int() { return static_cast<int>(*value); }
```

Comment:
This class is `FloatType`
Why would you create an `operator int()` function?
- [ ] revise